### PR TITLE
docs: add onboarding reproduction log entry (RankZephyr/FirstMistral DL20) via Colab CLI

### PR DESCRIPTION
**Pull request by k22mitta**
# Pull Request Checklist

## Reference Issue

- N/A

## Summary

- Reproduced onboarding results (RankZephyr + FirstMistral, DL20) using the Colab CLI recipe from #408, on an A100. Added a troubleshooting section for issues hit along the way.

## Why

Onboarding requires new students to try reranking using RankZephyr (with SPLADE retrieval) and FIRST via rankllm and record their results in the reproduction log. Also validates the Colab CLI recipe (#408) since Jimmy asked in #jarminions if someone could try it for students without local GPUs.

## Validation

Commands I ran:
GPU=A100 MODEL_PATH=castorini/rank_zephyr_7b_v1_full DATASET=dl20 RETRIEVAL_METHOD=SPLADE++_EnsembleDistil_ONNX TOP_K=100 CONTEXT_SIZE=4096 NUM_GPUS=1 EXTRA_RERANK_ARGS="--variable-passages" bash colab/rerank_on_colab.sh
GPU=A100 MODEL_PATH=castorini/first_mistral DATASET=dl20 RETRIEVAL_METHOD=SPLADE++_EnsembleDistil_ONNX TOP_K=100 CONTEXT_SIZE=4096 NUM_GPUS=1 EXTRA_RERANK_ARGS="--variable-passages --use-logits --use-alpha" bash colab/rerank_on_colab.sh
Results: ndcg_cut_10 = 0.7961 (RankZephyr), 0.7897 (FirstMistral).

## Follow-ups

- N/A

## Checklist Items

Before submitting your pull request, please review these items:

- [x] Have you followed the [contributing guidelines](https://github.com/castorini/rank_llm/blob/main/CONTRIBUTING.md)?
- [x] Have you verified that there are no existing [Pull Requests](https://github.com/castorini/rank_llm/pulls) for the same update/change?
- [x] Have you updated any relevant documentation or added new tests where needed?

# PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no API changes)
- [x] Documentation content changes
- [x] Reproduction logs
- [ ] Other...
  * Description:

### CODE REVIEW

**Comment by k22mitta** [comment]
Good catch — reworded that section to make clear the script is from the not-yet-merged #408, not something already in this tree. Fixed in 966d2cb.

**Comment by lintool** [comment]
Actually, closing... the exercises have changed so much that this is no longer relevant. Please try lesson again from scratch...
